src/Symfony/Component/Config/Definition/PrototypedArrayNode.php

```diff
@@ -156,6 +156,8 @@ public function getPrototype(): PrototypeNodeInterface
     /**
      * Disable adding concrete children for prototyped nodes.
      *
+     * @return never
```

Comment:
I think we should use `@return void` here, and maybe review where we added those and revert them
the reason is that if the class is truely extensible, then we have no reason to force child classes to also throw here

Reply:
We can surely challenge whether this should not be an `@final` class, but I think also without it we should force new implementations to not have children. It does not make sense for a prototyped node to contain concrete children.

Reply:
This should indeed be `never` for that one (it overrides the parent method in a similar way we do in FrozenParameterBag overriding ParameterBag methods)